src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.ExtensionMethodSymbolComputer.cs

```diff
@@ -35,11 +36,27 @@ private class ExtensionMethodSymbolComputer
             // The value indicates if we can reduce an extension method with this receiver type given receiver type.
             private readonly ConcurrentDictionary<ITypeSymbol, bool> _checkedReceiverTypes;
 
-            private static readonly AsyncBatchingWorkQueue<Project> _workQueue = new(
-                   TimeSpan.FromSeconds(1),
-                   BatchUpdateCacheAsync,
-                   AsynchronousOperationListenerProvider.NullListener,
-                   CancellationToken.None);
+            private static readonly object _gate = new();
+            private static AsyncBatchingWorkQueue<Project>? _workQueue;
+
+            private static AsyncBatchingWorkQueue<Project> GetWorkQueue(Project project)
+            {
+                lock (_gate)
+                {
+                    if (_workQueue is null)
+                    {
+                        var exportProvider = (IMefHostExportProvider)project.Solution.Workspace.Services.HostServices;
+                        var listenerProvider = exportProvider.GetExports<AsynchronousOperationListenerProvider>().Single().Value;
```

Comment:
📝 Code shouldn't assume `HostServices` implements `IMefHostExportProvider`

Reply:
You are commenting on an out-of-date snapshot, this is moved to https://github.com/dotnet/roslyn/pull/59760/files#diff-91071542ca3996ad4d7178bc276d40deaae3e2a3b7522531f29d869e1fb636ccR48

That said, I'm still using the same code to fet listener provider there. I assume I can still import the provider in the constructor safely?

Reply:
Fixed